tests/test_utilities.py

```diff
@@ -797,6 +797,101 @@ def test_qaoa_probabilities(self):
+
+    def test_delete_keys_from_dict(self):
+        """
+        Tests the function that deletes a set of keys from a dictionary: delete_keys_from_dict.
+        """
+
+        # Input dictionaries
+        input_dicts = [{'0':2/3, '1':1/3}, {'00':0,'01':0,'10':1/2,'11':1/2},
+                                {'000':0,'001':1/3,'010':0,'100':0,'011':1/3,'101':0,'110':1/3,'111':0},
+                                [{'list_0':1/3, 'list_1':1/3}, {'list_0':1/3, 'list_1':1/3}, {'list_0':1/3, 'list_1':1/3, 'list_2':1/3, 'list_3':1/3}]]
+
+        # Keys to be deleted
+        keys = ['0','00','000', 'list_0']
+
+        # Delete keys from dictionaries
+        output_dicts = delete_keys_from_dict(input_dicts,keys)
+
+        # expected output dictionaries
+        expected_dicts = [{'1':1/3}, {'01':0,'10':1/2,'11':1/2},
+                                {'001':1/3,'010':0,'100':0,'011':1/3,'101':0,'110':1/3,'111':0},
+                                [{'list_1':1/3}, {'list_1':1/3}, {'list_1':1/3, 'list_2':1/3, 'list_3':1/3}]]
+
+        # Test that each dictionary has been generated correctly
+        assert output_dicts == expected_dicts, f'Keys have not been deleted correctly'
+
+    def test_convert2serialize(self):
+        """
+        Tests the function that converts an object into a serializable dictionary: convert2serialize.
+        """
+
+        # define a new class
+        class TestClass:
+            def __init__(self, a, b):
+                self.a = a
+                self.b = b
+
+        # define a new class with _ast() method
+        class TestClass2:
+            def __init__(self, a, b):
+                self.a = a
+                self.b = b
+
+            def _ast(self):
+                return {'a':self.a,'b':self.b}
+
+        # Create an instance of the class
+        test_instance = TestClass(a = 1, b = 2)
+
+        # Convert the instance into a serializable dictionary
+        serialized_dict = convert2serialize(test_instance)
+
+        # Expected dictionary
+        expected_dict = {'a': 1, 'b': 2}
+
+        # Test that the dictionary has been generated correctly
+        assert serialized_dict == expected_dict, f'Object has not been converted correctly'
+
+        # now test with a list of instances, with attributes being list, dictionaries, complex numbers and numpy arrays
+        test_instance_list = [
+            TestClass(a = [1,2,3], b = {'x':1,'y':2}), TestClass2(a = np.array([1,2,3]), b = 1+1j)
+        ]
+
+        # Convert the instance into a serializable dictionary
+        serialized_dict = convert2serialize(test_instance_list, complex_to_string=True)
+
+        # Expected dictionary
+        expected_dict = [
+            {'a': [1,2,3], 'b': {'x':1,'y':2}}, {'a': [1,2,3], 'b': '(1+1j)'}
+        ]
+
+        # Test that the dictionary has been generated correctly
+        assert serialized_dict == expected_dict, f'Object has not been converted correctly'
+
+    def test_generate_uuid(self):
```

Comment:
We should also test:
- That all QAOA belonging to a RQAOA have the underlying RQAOA id
- Having assigned a project_id (or parent id) all children correctly feature it

Reply:
this we should test it, depending on how we decide to dump to the database, as I was saying here https://github.com/entropicalabs/openqaoa/pull/144#discussion_r1060344210 , no?